### PR TITLE
Docs/improve benchmarks and readme

### PR DESCRIPTION
**Pull request by pablocael**
- Added cpu model that generated benchmark results in README
- Improved README structure a bit

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,30 +2,30 @@
 
 Pyvptree is a python library, internally built in C++, for efficient KNN search using metric distance function such as L2 distance (see VPTreeL2Index) or Hamming distances (VPTreeBinaryIndex). 
 
-## Theoretical advantage of Vantage Points Trees compared to Kd-Trees
+## How VP-Trees work
 
-Intrinsic Dimensionality: VP-trees perform well in data with high intrinsic dimensionality, where the effective dimensionality is high. In contrast, kd-trees are efficient in low-dimensional spaces but their performance degrades quickly as the number of dimensions increases. This is often referred to as the "curse of dimensionality".
+VP-Trees are binary trees that successively divides spaces in order to peform different types of tasks, such as Nearest Neighbor Search. It differs from Kd-Trees in the sense that they always partition the whole space, instead of invidiual dimensional axes, using a specific metric function and a selected "Vantage Point" that will be used as reference to allow spliting the dataset. For more details on how it works please access the following references:
```
Comment (Dobatymo) [inline README.md:7]: divides spaces->divide space

File: README.md [hunk 2]
```diff
@@ -2,30 +2,30 @@
...
+- Metric Spaces: VP-trees can handle general metric spaces (any space where a distance function is defined that satisfies the triangle inequality), not just Euclidean spaces. This makes them more adaptable to different problem settings where the distance metric might not be the standard Euclidean distance.
 
-Also, practical implementation approaches such as using SIMD instructions or using highly optimized distance functions can strongly affect final performance in high dimensions. For instance, [Faiss Library](https://github.com/facebookresearch/faiss) performs very efficientlyin very high dimensions, where searches become near linear and exaustive, due to highly optimized code.
-
-Tipically spatial search structures tend to perform worse with increasing number of dimensions of dataset.
+- Handling Categorical Data: VP-trees, due to their use of arbitrary distance functions, can handle categorical data more naturally than kd-trees. While there are workarounds to use kd-trees with categorical data, they often require substantial tweaking and may not perform optimally.
```
Comment (Dobatymo) [inline README.md:21]: maybe for discrete metric spaces we can mention BK-trees here

File: README.md [hunk 3]
```diff
@@ -2,30 +2,30 @@
...
 
+- Balanced Tree Structure: VP-trees inherently try to create a balanced tree structure, which is beneficial for efficient searching. kd-trees can become unbalanced in certain situations, particularly with un-uniform data, leading to inefficient search operations.
```
Comment (Dobatymo) [inline README.md:23]: un-uniform -> non-uniform

File: README.md [hunk 4]
```diff
@@ -2,30 +2,30 @@
...
 
-# How this library works
+Its importante to notice, however, that practical implementation approaches such as using [SIMD](https://en.wikipedia.org/wiki/Single_instruction,_multiple_data) instructions or using highly optimized distance functions can strongly affect final performance in high dimensions. For instance, [Faiss Library](https://github.com/facebookresearch/faiss) performs very efficientlyin very high dimensions, where searches become near linear and exaustive, due to highly optimized code.
```
Comment (Dobatymo) [inline README.md:25]: Its importante -> It's important, efficientlyin -> efficiently in

File: README.md [hunk 5]
```diff
@@ -2,30 +2,30 @@
...
 
-This library implements a [Vantage Point Tree](https://en.wikipedia.org/wiki/Vantage-point_tree) to perform search within multidimensional metric spaces using arbitrary distance functions.
+Tipically spatial search structures tend to perform worse with increasing number of dimensions of dataset.
```
Comment (Dobatymo) [inline README.md:27]: Tipically -> Typically

**Review by Dobatymo — CHANGES_REQUESTED** [review]
mostly typos

File: README.md (hunk 5 above)
Comment (pablocael) [reply README.md]: Done on 67eff0da7685a720e2a4ce15f2f821a11e78b322

File: README.md (hunk 4 above)
Comment (pablocael) [reply README.md]: Done on 67eff0da7685a720e2a4ce15f2f821a11e78b322

File: README.md (hunk 3 above)
Comment (pablocael) [reply README.md]: Done on 67eff0da7685a720e2a4ce15f2f821a11e78b322

File: README.md (hunk 2 above)
Comment (pablocael) [reply README.md]: Done on 67eff0da7685a720e2a4ce15f2f821a11e78b322

File: README.md (hunk 1 above)
Comment (pablocael) [reply README.md]: Done on 67eff0da7685a720e2a4ce15f2f821a11e78b322

File: README.md [hunk 6]
```diff
@@ -18,13 +18,13 @@ VP-Trees are binary trees that successively divides spaces in order to peform di
...
 
-Its importante to notice, however, that practical implementation approaches such as using [SIMD](https://en.wikipedia.org/wiki/Single_instruction,_multiple_data) instructions or using highly optimized distance functions can strongly affect final performance in high dimensions. For instance, [Faiss Library](https://github.com/facebookresearch/faiss) performs very efficientlyin very high dimensions, where searches become near linear and exaustive, due to highly optimized code.
+Its important to notice, however, that practical implementation approaches such as using [SIMD](https://en.wikipedia.org/wiki/Single_instruction,_multiple_data) instructions or using highly optimized distance functions can strongly affect final performance in high dimensions. For instance, [Faiss Library](https://github.com/facebookresearch/faiss) performs very efficiently in very high dimensions, where searches become near linear and exaustive, due to highly optimized code.
```
Comment (Dobatymo) [inline README.md:25]: Its -> It's

**Review by Dobatymo — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (pablocael) [reply README.md]: Thanks Fixed on b83641362576df140afc406edd91950e50eef713

**Review by Dobatymo — APPROVED** [review]

**Pull request merged**